src/views/auth/auth.js

```diff
@@ -0,0 +1,27 @@
+import React, { PropTypes, Component } from 'react';
+
+export default class Auth extends Component {
+
+    static propTypes = {
+        children: PropTypes.node.isRequired,
+        loggedIn: PropTypes.bool.isRequired,
+        checkAuth: PropTypes.func.isRequired,
+        redirect: PropTypes.string.isRequired,
+        history: PropTypes.object.isRequired,
+    };
+
+    componentDidMount() {
+        const { loggedIn, checkAuth, redirect, history } = this.props;
+        if (!loggedIn) {
+            checkAuth(redirect, history);
+        }
+    }
+
+    render() {
+        const { loggedIn } = this.props;
+        if (loggedIn) {
+            return this.props.children;
```

Comment:
1) Theoretically, couldn't this method just be `return this.props.children;` b/c `componentDidMount` will always run before `render`, and if the user is not logged in they will always redirect to the login page?

2) When will we hit the auth-check endpoint through this component?  Every route/component is `index.js` change?

3) We need to talk about including the auth-token in all requests made against the API -- I assume we'll still need to even when doing it this way.

3.1) Consider this: the user uses the application for a while, then walks away in which time the token expires.  The user comes back and tries to do something (EG update a tag).  The endpoint returns a bad authentication response.  How can we handle this?  Basically, we need a "failure" `then` block to all API requests to check for this and handle appropriately -- seems like a good reason to refactor the `$.ajax` function out to a util like we were earlier this week.